sktime/forecasting/base/adapters/_darts.py

```diff
@@ -0,0 +1,241 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements adapter for Darts models."""
+import abc
+from typing import List, Optional
+
+import pandas
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+__author__ = ["yarnabrina"]
+
+
+class _DartsAdapter(BaseForecaster):
+    """Base adapter class for Darts models.
+
+    Parameters
+    ----------
+    past_covariates : Optional[List[str]], optional
+        column names in ``X`` which are known only for historical data, by default None
+    num_samples : Optional[int], optional
+        Number of times a prediction is sampled from a probabilistic model, by default
+        1000
+
+    Notes
+    -----
+    If unspecified, all columns will be assumed to be known during prediction duration.
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "enforce_index_type": pandas.DatetimeIndex,
```

Comment:
@fkiraly Can you please tell if this will be ensured during tests, or tests can pass datasets with any index type leading to failures?

Reply:
Not sure - I think the tag has no effect. There is an issue somewhere that states this, and asks to implement coercion.

Reply:
Ok, found it https://github.com/sktime/sktime/issues/3641.

What will be better to do here? Fix this issue, or add an if-else to check index type and raise TypeError? Or, both?

Reply:
yes, that was it

> What will be better to do here?

I would do it in a separate PR, but yes.

I also would strongly recommend to think carefully about consequences of starting to raise an exception - this may cause estimators that have previously "over-restricted" the index type to fail, so it may break working user code!

I've added more details with my now-preferred solution here: https://github.com/sktime/sktime/issues/3641#issuecomment-1667237858 (this is based on making `Prophet` work for any index type)